mindmeld/models/text_models.py

```diff
@@ -472,5 +383,124 @@ def _get_feature_scaler(self):
+
+        evaluations = [
+            EvaluatedExample(
+                e, labels[i], predictions[i][0], predictions[i][1], config.label_type
+            )
+            for i, e in enumerate(examples)
+        ]
+
+        model_eval = StandardModelEvaluation(config, evaluations)
+        return model_eval
+
+    def fit(self, examples, labels, params=None):
+        """Trains this model.
+
+        This method inspects instance attributes to determine the classifier
+        object and cross-validation strategy, and then fits the model to the
+        training examples passed in.
+
+        Args:
+            examples (ProcessedQueryList.*Iterator): A list of examples.
+            labels (ProcessedQueryList.*Iterator): A parallel list to examples. The gold labels
+                for each example.
+            params (dict, optional): Parameters to use when training. Parameter
+                selection will be bypassed if this is provided
+
+        Returns:
+            (TextModel): Returns self to match classifier scikit-learn \
+                interfaces.
+        """
+        params = params or self.config.params
+        skip_param_selection = params is not None or self.config.param_selection is None
+
+        # Shuffle to prevent order effects
+        indices = list(range(len(labels)))
+        random.shuffle(indices)
+        examples.reorder(indices)
+        labels.reorder(indices)
+        distinct_labels = set(labels)
+        if len(set(distinct_labels)) <= 1:
+            return self
+
+        # Extract features and classes
+        y = self._label_encoder.encode(labels)
+        X, y, groups = self.get_feature_matrix(examples, y, fit=True)
+
+        if skip_param_selection:
+            self._clf = self._fit(X, y, params)
+            self._current_params = params
+        else:
+            # run cross validation to select params
+            best_clf, best_params = self._fit_cv(X, y, groups)
+            self._clf = best_clf
+            self._current_params = best_params
+
+        return self
+
+    def predict(self, examples, dynamic_resource=None):
+        X, _, _ = self.get_feature_matrix(examples, dynamic_resource=dynamic_resource)
+        y = self._clf.predict(X)
+        predictions = self._class_encoder.inverse_transform(y)
+        return self._label_encoder.decode(predictions)
+
+    def predict_proba(self, examples, dynamic_resource=None):
+        X, _, _ = self.get_feature_matrix(examples, dynamic_resource=dynamic_resource)
+        return self._predict_proba(X, self._clf.predict_proba)
+
+    def view_extracted_features(self, example, dynamic_resource=None):
+        return self._extract_features(
+            example, dynamic_resource=dynamic_resource, tokenizer=self.tokenizer
+        )
+
+    def dump(self, path, metadata=None):
+        metadata = metadata or {}
+        metadata.update({
+            "model": self,
+            "model_config": self.config,
+            "serializable": True}
```

Comment:
Why is `serializable ` field necessary? What models are not serializable?

Reply:
Since we have different base classes for classical and neural models, do we still need this key?

Reply:
We just need it in `TaggerModel`, as it currently also supports LSTM model.